allow modification of Replica1 type to Replica2 or Replica3

amarts opened the pull request:
Along with it, fix a bug which resulted in array index out of range.

Signed-off-by: Amar Tumballi <amar@kadalu.io>

amarts committed:
allow modification of Replica1 type to Replica2 or Replica3

Along with it, fix a bug which resulted in array index out of range.

Signed-off-by: Amar Tumballi <amar@kadalu.io>

leelavg commented on operator/main.py:
```diff
@@ -604,7 +595,8 @@ def handle_modified(core_v1_client, obj):
     # Set Node ID for each storage device from configmap
     for idx, _ in enumerate(obj["spec"]["storage"]):
-        obj["spec"]["storage"][idx]["node_id"] = cfgmap["bricks"][idx]["node_id"]
+        if not obj["spec"]["storage"][idx].get("node_id", None):
```
Will it always be None? (or) something like `node_id = cfgmap["bricks"][idx].get("node_id") or str(uuid.uuid1())` would be better ig.

amarts replied:
The bug fixed here is to not have `idx` on cfgmap['bricks'].

The usecase is this: I had a config with type ReplicaX/Disperse., with say M nodes. Now, I added X nodes, and did `kubeclt apply`. At this time, cfgmap entry would have say M entries, and obj would have M+X entries. Hence the out of range. 

But thinking more on this, everytime the config changes, the node_id changes. But considering it gets used in server config properly as server pods gets restarted, it is OK for us for now.

leelavg reviewed (commented):
Minor comment

amarts committed:
operator: set node-id as simple idx instead of uuid

This reduces possible config change for existing servers, which
means, they can avoid a restart.

Signed-off-by: Amar Tumballi <amar@kadalu.io>

leelavg reviewed (approved):
LGTM
